source/common/network/listen_socket_impl.cc

```diff
@@ -48,10 +48,12 @@ void ListenSocketImpl::setupSocket(const Network::Socket::OptionsSharedPtr& opti
 template <>
 void NetworkListenSocket<
     NetworkSocketTrait<Address::SocketType::Stream>>::setPrebindSocketOptions() {
-  // TODO(htuch): This might benefit from moving to SocketOptionImpl.
+
   int on = 1;
-  int rc = setsockopt(fd_, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on));
-  RELEASE_ASSERT(rc != -1, "");
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
```

Comment:
drive-by comment: Can we plumb the Api::Api& through to here or do we need to depend on a static?

Reply:
...I just noticed that this was already approved by @htuch so feel free to ignore for now...but maybe this could be a follow-up.

Reply:
@jmarantz historically we were cool with this singleton for OS syscalls, since it seemed a pain to plumb Api everywhere just to get a libc mock capability. Have we decided to phase that out?